src/image.cpp

```diff
@@ -242,10 +264,18 @@ void image::threshold(bool otsu)
     }
 }
 
-image image::operator+(const image& other) const
+image image::operator+(image other) const &
+{
+    assert(rows() == other.rows());
+    cv::Mat result;
+    cv::hconcat(m_data, std::move(other.m_data), result);
+    return image(std::move(result));
+}
+
+image image::operator+(image other) &&
 {
     assert(rows() == other.rows());
     cv::Mat result;
-    cv::hconcat(m_data, other.m_data, result);
+    cv::hconcat(std::forward<cv::Mat>(m_data), std::move(other.m_data), result);
     return image(std::move(result));
```

Comment:
Not sure about this.

- `&&`-qualification implies move, this is not a forwarding reference
- I see a way to append rows in the `Mat` API, but it looks like there is no way to append columns which would do the horizontal contactenation. If there is no efficient _happend_, then a `&&` qualified member will be as slow as the normal one. If I've missed that API, this mem function could become simply:

```
m_data.fancy_function_that_appends_columns(std::move(other.m_data));
return image(std::move(m_data));
```